tests/unit/test_repository.py

```diff
@@ -265,6 +267,26 @@ def test_registering_a_new_stack_with_already_registered_components(clean_repo):
     )
 
 
+def test_updating_a_new_stack_with_already_registered_components(clean_repo):
```

Comment:
I would suggest maybe testing in the stack store tests, because those are parameterized to test every implementation, whereas this will only test the yaml version I believe.

Reply:
Good idea. I've moved it over.